Docs: Clarify data streaming with plugins documentation (#35576)
grafana/grafana: docs/sources/live/live-feature-overview.md

## Data streaming from plugins

> With a help of Grafana Live data source plugins can stream data updates in the form of Grafana data frames to a frontend.

Reviewer: ```suggestion
With Grafana Live, backend data source plugins can stream updates to frontend panels.
```

maybe?

Reviewer 2: @FZambia, does this suggestion clarify things?

Author: Yes, previously reader could think that there is some sort of `Grafana Live data source` which is confusing.
